bindings/python/setup.py

```diff
@@ -49,78 +86,84 @@ def append_libbson_flags(module):
         orig_ldflags = os.environ.get('LDFLAGS', '')
         os.environ['LDFLAGS'] = ldflags + " " + orig_ldflags
 
-    # Ensure our Cython extension can dynamically link to vendored libbson
-    # - https://blog.krzyzanowskim.com/2018/12/05/rpath-what/
-    # - https://nehckl0.medium.com/creating-relocatable-linux-executables-by-setting-rpath-with-origin-45de573a2e98
-    if platform == "darwin":
-        module.extra_link_args += ["-rpath", "@loader_path"]
-    elif platform == 'linux':
-        module.extra_link_args += ["-Wl,-rpath,$ORIGIN"]
-
     # https://cython.readthedocs.io/en/latest/src/tutorial/external.html#dynamic-linking
-    lnames = lnames.split()
     # Strip whitespace to avoid weird linker failures on manylinux images
-    libnames = [lname.lstrip('-l').strip() for lname in lnames]
+    libnames = [lname.lstrip('-l').strip() for lname in lnames.split()]
     module.libraries.extend(libnames)
 
 
 def append_arrow_flags(module):
     import numpy as np
     import pyarrow as pa
 
-    module.include_dirs.append(np.get_include())
-    module.include_dirs.append(pa.get_include())
-    module.library_dirs.extend(pa.get_library_dirs())
+    if os.name == 'nt':
+        module.include_dirs.append(np.get_include())
+        module.include_dirs.append(pa.get_include())
+    else:
+        module.extra_compile_args.append("-isystem" + pa.get_include())
+        module.extra_compile_args.append("-isystem" + np.get_include())
```

Comment:
What's this for? Is this a replacement for:
```
    module.include_dirs.append(np.get_include())
    module.include_dirs.append(pa.get_include())
```

Reply:
Yes, and also what `snowflake-connector-python` used